packages/react/src/ReactHooks.js

```diff
@@ -121,6 +121,9 @@ export function useEffect(
   create: () => (() => void) | void,
   deps: Array<mixed> | void | null,
 ): void {
+  if (typeof deps === "undefined") {
+    console.warn("Missing dependency array in useEffect. This will lead to an infinite loop. To fix, please add a dependency array to useEffect. Docs: https://react.dev/reference/react/useEffect");
```

Comment:
Please take into consideration the following remarks:
- This warning will be printed even if useEffect is called outside react
- This warning will be printed each time useEffect is called, not only the first time
- It doesn't indicate the component that's impacted (its name/filename)
- This warning, if applied, should be part of `mountFiber` in ReactFiberHooks.js and only under __DEV__

Moreover, I don't believe the dependencies array are the solution to infinite loops, but rather the state update that occurs in it, and mostly an if statement solves the issue (I'm implying that you can solve infinite loops even in effects where there is no dependencies array at all).
Also, I believe there are use cases when calling an effect at each render is okay, such as:
```
let subscription = retain(store, forceUpdate, ...); // created evey render
// commits the subscription to store.listeners
useEffect(commitSubscription.bind(null, subscription));
```

An equivalent code involves using useMemo with a fixed deps array, and useEffect with other dependencies, which is not necessarily better, since it will involve closures (and obviously state ones) + a bunch of checks to perform every time.